internal/config/namespace.go

```diff
@@ -31,8 +34,22 @@ var (
 	// this list is removed from all managed namespaces unless specifically specified by the HC of the
 	// namespace or one of its ancestors.
 	managedAnnotations []*regexp.Regexp
+
+	// hncNamespace is the namespace where hnc-manager and hnc resources deployed. It set by commandline argument,
+	// default to hnc-system.
+	hncNamespace string
 )
 
+var (
+	once sync.Once
+)
+
+func SetHncNamespace(ns string) {
+	once.Do(func() {
```

Comment:
I think this is overkill - I'd just say `hncNamespace = ns` and not bother with the sync stuff. None of the other `Set` functions in this package go to that level of paranoia :)